x/gamm/types/errors.go

```diff
@@ -4,16 +4,17 @@ import sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 
 // x/gamm module sentinel errors.
 var (
-	ErrPoolNotFound       = sdkerrors.Register(ModuleName, 1, "pool not found")
-	ErrPoolAlreadyExist   = sdkerrors.Register(ModuleName, 2, "pool already exist")
-	ErrPoolLocked         = sdkerrors.Register(ModuleName, 3, "pool is locked")
-	ErrTooFewPoolAssets   = sdkerrors.Register(ModuleName, 4, "pool should have at least 2 assets, as they must be swapping between at least two assets")
-	ErrTooManyPoolAssets  = sdkerrors.Register(ModuleName, 5, "pool has too many assets (currently capped at 8 assets per balancer pool and 2 per stableswap)")
-	ErrLimitMaxAmount     = sdkerrors.Register(ModuleName, 6, "calculated amount is larger than max amount")
-	ErrLimitMinAmount     = sdkerrors.Register(ModuleName, 7, "calculated amount is lesser than min amount")
-	ErrInvalidMathApprox  = sdkerrors.Register(ModuleName, 8, "invalid calculated result")
-	ErrAlreadyInvalidPool = sdkerrors.Register(ModuleName, 9, "destruction on already invalid pool")
-	ErrInvalidPool        = sdkerrors.Register(ModuleName, 10, "attempting to create an invalid pool")
+	ErrPoolNotFound        = sdkerrors.Register(ModuleName, 1, "pool not found")
+	ErrPoolAlreadyExist    = sdkerrors.Register(ModuleName, 2, "pool already exist")
+	ErrPoolLocked          = sdkerrors.Register(ModuleName, 3, "pool is locked")
+	ErrTooFewPoolAssets    = sdkerrors.Register(ModuleName, 4, "pool should have at least 2 assets, as they must be swapping between at least two assets")
+	ErrTooManyPoolAssets   = sdkerrors.Register(ModuleName, 5, "pool has too many assets (currently capped at 8 assets per balancer pool and 2 per stableswap)")
+	ErrLimitMaxAmount      = sdkerrors.Register(ModuleName, 6, "calculated amount is larger than max amount")
+	ErrLimitMinAmount      = sdkerrors.Register(ModuleName, 7, "calculated amount is lesser than min amount")
+	ErrInvalidMathApprox   = sdkerrors.Register(ModuleName, 8, "invalid calculated result")
+	ErrAlreadyInvalidPool  = sdkerrors.Register(ModuleName, 9, "destruction on already invalid pool")
+	ErrInvalidPool         = sdkerrors.Register(ModuleName, 10, "attempting to create an invalid pool")
+	ErrDenomNotFoundInPool = sdkerrors.Register(ModuleName, 11, "denom does not exist in pool")
```

Comment:
Added this

Reply:
oh wow this is actually a pretty good implementation with the second interface we talked about we should definitely use this PR in the future as reference